### PR TITLE
Implemented role tags

### PR DESCRIPTION
**Pull request by cyclane**
Implemented [role tags](https://discord.com/developers/docs/topics/permissions#role-object-role-tags-structure) into the Role class as attributes `botID`, `integrationID` and `premiumSubscriber`.
- Converted the premiumSubscriber to a boolean which makes more sense than keeping it as null type

### CODE REVIEW

File: lib/structures/Role.js [hunk 1]
```diff
@@ -92,13 +110,16 @@ class Role extends Base {
 
     toJSON(props = []) {
         return super.toJSON([
+            "botID",
             "color",
             "hoist",
+            "integrationID ",
```
Comment (bsian03) [inline lib/structures/Role.js]:
```suggestion
            "integrationID",
```

File: lib/structures/Role.js [hunk 2]
```diff
@@ -92,13 +110,16 @@ class Role extends Base {
...
             "managed",
             "mentionable",
             "name",
             "permissions",
             "position",
+            "premiumSubscriber ",
```
Comment (bsian03) [inline lib/structures/Role.js]:
```suggestion
            "premiumSubscriber",
```

**Review by bsian03 — CHANGES_REQUESTED** [review]
Extra spaces you added here

**Comment by cyclane** [comment]
Thank you for pointing those out, did not notice I added them

File: lib/structures/Role.js [hunk 3]
```diff
@@ -47,6 +50,21 @@ class Role extends Base {
...
         }
+        if(data.tags !== undefined) {
+            if(data.tags.bot_id !== undefined) {
+                this.botID = data.tags.bot_id;
+            }
+            if(data.tags.integration_id !== undefined) {
+                this.integrationID = data.tags.integration_id;
+            }
+            if(data.tags.premium_subscriber !== undefined) {
+                this.premiumSubscriber = true;
```
Comment (Khaaz) [inline lib/structures/Role.js:61]: any particular reason for making this a top level field instead of being consistent with discord here?

**Review by abalabahaha — CHANGES_REQUESTED** [review]
Agreed, more of a fan of preserving the tags object for future proofing

**Comment by abalabahaha** [comment]
Implemented in 824ccf7

**Pull request closed without merging**